### PR TITLE
refactor(core): Refactor withHealthchecks into infra-next-server

### PR DESCRIPTION
**Pull request by saevarma**
## What

auth-admin-web was using `withHealthchecks` from the web project, creating dependency that shouldn't exist.
Also each next app has copy-paste of the `withHealthchecks` wrapper and the config.

This PR refactors the `withHealthchecks` into the `infra-next-server`

## Why

- Decouple unwanted dependency.
- Make withHealthchecks part of infra-next-server so individuals apps only needs to configure external endpoint dependcies.

## Screenshots / Gifs

N/A

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: workspace.json [hunk 1]
```diff
@@ -9510,7 +9510,7 @@
             "main": "apps/web/server.ts",
             "tsConfig": "apps/web/tsconfig.json",
             "maxWorkers": 2,
-            "assets": ["apps/web/next-modules"]
```
Comment (eirikurn) [inline workspace.json:9513]: Such good feelings :D

**Review by eirikurn — APPROVED** [review]
Sooo much nicer! Great job! 💯

File: libs/infra-next-server/src/lib/health.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    Readiness should return 200 if it can resolve hostnames of external dependencies.
+    Apps should inform the cluster of their own state not the state of other services.
+    If we make this app depend on external service we risk it going down with those external dependecies.
+    e.g the web app should render the error page for users if the api goes offline the web app should not go offline
+    */
+
+    const dependencies =
+      typeof externalEndpointDependencies === 'function'
+        ? externalEndpointDependencies()
+        : externalEndpointDependencies
```
[the remark this answers is not in the document]
Comment (saevarma) [reply libs/infra-next-server/src/lib/health.ts:29]: Yes that is the reason. Passing in the nextjs config is a great idea to simplify those function.

Ahh ofc we shouldn't be calling it in the request handler, will move it.

**Review by Snojo — APPROVED** [review]
LGTM

**Review by essgje — APPROVED** [review]
left

**Review by kristofer-lp8f — CHANGES_REQUESTED** [review]
Looks good.

I'll approve upon comment resolution.

**Review by davidpxn — APPROVED** [review]
LGTM

File: libs/infra-next-server/src/lib/health.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    typeof externalEndpointDependencies === 'function'
+      ? externalEndpointDependencies(nextConfig)
+      : externalEndpointDependencies
+
+  app.use('/readiness', async (_, res) => {
+    /*
+    Readiness should return 200 if it can resolve hostnames of external dependencies.
+    Apps should inform the cluster of their own state not the state of other services.
+    If we make this app depend on external service we risk it going down with those external dependecies.
+    e.g the web app should render the error page for users if the api goes offline the web app should not go offline
```
[the remark this answers is not in the document]
Comment (saevarma) [reply libs/infra-next-server/src/lib/health.ts]: Added missing comma and period.

**Pull request merged**